sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/ColumnVector.java

```diff
@@ -970,30 +458,14 @@ public final int appendStruct(boolean isNull) {
   protected boolean anyNullsSet;
 
   /**
-   * True if this column's values are fixed. This means the column values never change, even
-   * across resets.
-   */
-  protected boolean isConstant;
-
-  /**
-   * Default size of each array length value. This grows as necessary.
-   */
-  protected static final int DEFAULT_ARRAY_LENGTH = 4;
-
-  /**
-   * Current write cursor (row index) when appending data.
-   */
-  protected int elementsAppended;
-
-  /**
    * If this is a nested type (array or struct), the column for the child data.
    */
   protected ColumnVector[] childColumns;
```

Comment:
can we move this to `WritableColumnVector`? I think `ColumnVector` only need `ColumnVector getChildColumn(int ordinal)`, and `WritableColumnVector` can overwrite it to `WritableColumnVector getChildColumn(int ordinal)`